dev/com.ibm.ws.jdbc_fat_v43/test-applications/app43/src/jdbc/fat/v43/web/JDBC43TestServlet.java

```diff
@@ -1669,6 +1789,93 @@ public void testSuspendRunLTCAndResumeUnsharable() throws Exception {
+            con1.setTransactionIsolation(Connection.TRANSACTION_REPEATABLE_READ);
+            String k1 = con1.getClientInfo("SHARDING_KEY"); // extension by mock JDBC driver to determine the sharding key
+            assertTrue(k1, k1.endsWith("|CHAR:XACBKey;BOOLEAN:true;"));
+
+            Connection con2 = conbuilderB.build();
+            // If connection handle is shared, it will report the isolation level value of con1,
+            assertEquals(Connection.TRANSACTION_REPEATABLE_READ, con2.getTransactionIsolation());
+            String k2 = con2.getClientInfo("SHARDING_KEY"); // extension by mock JDBC driver to determine the sharding key
+            assertTrue(k2, k2.endsWith("|CHAR:XACBKey;BOOLEAN:true;"));
+
+            PreparedStatement ps2 = con2.prepareStatement("INSERT INTO STREETS VALUES (?, ?, ?)");
+            ps2.setString(1, "Plummer Circle SW");
+            ps2.setString(2, "Rochester");
+            ps2.setString(3, "MN");
+            ps2.executeUpdate();
+            ps2.close();
+
+            requests2 = (AtomicInteger[]) con2.unwrap(Supplier.class).get();
+            begins2 = requests2[BEGIN].get();
+
+            // enlist another resource to prove that the prior connection is two-phase capable
+            Connection con3 = sharablePool1DataSourceWithAppAuth.getConnection("user43", "pwd43");
+            PreparedStatement ps3 = con3.prepareStatement("INSERT INTO STREETS VALUES (?, ?, ?)");
+            ps3.setString(1, "Rocky Creek Drive NE");
+            ps3.setString(2, "Rochester");
+            ps3.setString(3, "MN");
+            ps3.executeUpdate();
+            ps3.close();
+
+            con3.close();
+            con2.close();
+            con1.close();
+        } finally {
+            tx.commit();
+        }
+
+        // Specify a similar, but different (subkeys are reversed) sharding key on the builder,
+        ShardingKeyBuilder keybuilderD = sharableXADataSource.createShardingKeyBuilder();
+        ShardingKey key4 = keybuilderD.subkey(true, JDBCType.BOOLEAN).subkey("XACBKey", JDBCType.CHAR).build();
+        Connection con4 = conbuilderA.shardingKey(key4).build();
+        try {
+            String k4 = con4.getClientInfo("SHARDING_KEY"); // extension by mock JDBC driver to determine the sharding key
+            assertTrue(k4, k4.endsWith("|BOOLEAN:true;CHAR:XACBKey;"));
+        } finally {
+            con4.close();
+        }
+
+        // Specify a match for the first sharding key
+        ShardingKey key5 = keybuilderB.build();
+        Connection con5 = conbuilderB.shardingKey(key5).build();
+        try {
+            assertEquals(Connection.TRANSACTION_READ_COMMITTED, con5.getTransactionIsolation());
+            String k5 = con5.getClientInfo("SHARDING_KEY"); // extension by mock JDBC driver to determine the sharding key
+            assertTrue(k5, k5.endsWith("|CHAR:XACBKey;BOOLEAN:true;"));
```

Comment:
Weren't there no subkeys specified on this connection?

Reply:
The subkeys were set upon previous usage of the builder:
ShardingKey keyB = keybuilderB.subkey("XACBKey", JDBCType.CHAR).subkey(true, JDBCType.BOOLEAN).build();
Note that the JDBC 4.3 spec is vague about what happens when a ShardingKeyBuilder is reused -- whether it rejects reuse, allows reuse with subkeys preserved, allows reuse with subkeys cleared, or something else.  Our mock JDBC driver happens to allow reuse with the subkeys preserved, which is the behavior that this test case relies on here.